judge/views/contests.py

```diff
@@ -327,6 +327,30 @@ def get_context_data(self, **kwargs):
 
         return context
 
+class ContestAllProblems(ContestMixin, TitleMixin, DetailView):
+    template_name = 'contest/contest-all-problems.html'
+
+    def get_title(self):
+        return self.object.name
+
+    def get_context_data(self, **kwargs):
+        context = super(ContestAllProblems, self).get_context_data(**kwargs)
+        context['contest_problems'] = Problem.objects.filter(contests__contest=self.object) \
```

Comment:
This seems to not check any permission, any user can access this page, right? So that is not safe.